modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java

```diff
@@ -46,26 +46,40 @@
 
 	private static Log log = LogFactory.getLog(OperationMgtService.class);
 
-	@GET
+	@PUT
 	@Produces(MediaType.APPLICATION_JSON)
 	@Path("{id}")
 	public List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> getPendingOperations
-			(@HeaderParam("Accept") String acceptHeader, @PathParam("id") String id) {
+			(@HeaderParam("Accept") String acceptHeader, @PathParam("id") String id,
+			 List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> resultOperations) {
 
+		if (log.isDebugEnabled()) {
+			log.debug("Invoking Android pending operations:" + id);
+		}
 		Message message = new Message();
 		MediaType responseMediaType = AndroidAPIUtils.getResponseMediaType(acceptHeader);
-		java.util.List<? extends org.wso2.carbon.device.mgt.common.operation.mgt.Operation> operations = new
-				ArrayList<Operation>();
-		if (log.isDebugEnabled()) {
-			log.debug("Invoking Android pending operations:"+id);
+
+		try {
+			if (resultOperations != null) {
+				updateOperations(resultOperations);
+			}
+		} catch (OperationManagementException e) {
+			message.setResponseMessage("Issue in retrieving operation management service instance");
```

Comment:
Maybe we should assign the error message to a proper String variable and pass it for logging as well as into the instance of "Message".